### PR TITLE
raftstore: gc queues

### PR DESCRIPTION
**Pull request by BusyJay**
When the queues are not used anymore, they can be shrink to a smaller one to save memory.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
rest LGTM

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -92,6 +94,13 @@ impl ReadIndexQueue {
             }
         }
     }
+
+    fn gc(&mut self) {
+        if self.reads.capacity() > SHRINK_CACHE_CAPACITY && self.reads.len() < SHRINK_CACHE_CAPACITY
```
[the remark this answers is not in the document]
Comment (hicqu) [reply src/raftstore/store/peer.rs]: We also considered capacity so that maybe it's not a problem?

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**